src/relayserver/ServerConfigParams.ts

```diff
@@ -0,0 +1,181 @@
+  gasPricePercent: 'number',
+  ethereumNodeUrl: 'string',
+  workdir: 'string',
+  devMode: 'boolean',
+  debug: 'boolean',
+  registrationBlockRate: 'number',
+
+  workerMinBalance: 'number',
+  workerTargetBalance: 'number',
+  managerMinBalance: 'number',
+  managerTargetBalance: 'number',
+  minHubWithdrawalBalance: 'number'
+
+} as any
+
+// by default: no waiting period - use VersionOracle entries immediately.
+const DefaultOracleDelayPeriod = 0
+
+// helper function: throw and never return..
+function error (err: string): never {
+  throw new Error(err)
+}
+
+// get the keys matching specific type from ConfigParamsType
+export function filterType (config: any, type: string): any {
+  return Object.entries(config).flatMap(e => e[1] === type ? [e[0]] : [])
+}
+
+// convert [key,val] array (created by Object.entries) back to an object.
+export function entriesToObj (entries: any[]): any {
+  return entries
+    .reduce((set: any, [k, v]) => ({ ...set, [k]: v }), {})
+}
+
+// filter and return from env only members that appear in "config"
+export function filterMembers (env: any, config: any): any {
+  return entriesToObj(Object.entries(env)
+    .filter(e => isDefined(config[e[0]])))
+}
+
+// map value from string into its explicit type (number, boolean)
+// TODO; maybe we can use it for more specific types, such as "address"..
+function explicitType ([key, val]: [string, any]): any {
+  const type = ConfigParamsTypes[key]
+  if (type === undefined) {
+    error(`unexpected param ${key}=${val}`)
+  }
+  switch (type) {
+    case 'boolean' :
+      if (val === 'true' || val === true) return [key, true]
+      if (val === 'false' || val === false) return [key, false]
+      break
+    case 'number': {
+      const v = parseInt(val)
+      if (!isNaN(v)) {
+        return [key, v]
+      }
+      break
+    }
+    default:
+      return [key, val]
+  }
+  error(`Invalid ${type}: ${key} = ${val}`)
+}
+
+/**
+ * initialize each parameter from commandline, env or config file (in that order)
+ * config file must be provided either as command-line or env (obviously, not in
+ * the config file..)
+ */
+export function parseServerConfig (args: string[], env: any): any {
+  const envDefaults = filterMembers(env, ConfigParamsTypes)
+
+  const argv = parseArgs(args, {
+    string: filterType(ConfigParamsTypes, 'string'),
+    // boolean: filterType(ConfigParamsTypes, 'boolean'),
+    default: envDefaults
+  })
+  if (argv._.length > 0) {
+    error(`unexpected param(s) ${argv._.join(',')}`)
+  }
+  delete argv._
+  let configFile = {}
+  if (argv.config != null) {
+    if (!fs.existsSync(argv.config)) {
+      error(`unable to read config file "${argv.config}"`)
+    }
+    configFile = JSON.parse(fs.readFileSync(argv.config, 'utf8'))
+  }
+  const config = { ...configFile, ...argv }
+  return entriesToObj(Object.entries(config).map(explicitType))
+}
+
+function isDefined (obj: any): boolean {
+  return obj !== null && obj !== undefined
```

Comment:
You can just check `obj != null` since `undefined == null`

Reply:
removed (last week) helper function, since it was unhelpful